### PR TITLE
Fix memory leaks in Spine caused by not calling freeStoreMemory.

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #
https://github.com/cocos/3d-tasks/issues/18640

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by dumganhar — APPROVED** [review]

File: native/cocos/editor-support/spine-wasm/spine-wasm.cpp [hunk 1]
```diff
@@ -96,16 +96,9 @@ void SpineWasmUtil::destroySpineSkeleton(Skeleton* skeleton) {
...
-        if (s_memSize < size) {
-            delete[] s_mem;
-            s_mem = new uint8_t[size];
-            s_memSize = size;
-        }
-    } else {
-        s_mem = new uint8_t[size];
-        s_memSize = size;
-    }
+    s_mem = new uint8_t[size];
```
Comment (minggo) [inline native/cocos/editor-support/spine-wasm/spine-wasm.cpp:98]: Why don't have to delete previous memory?

Comment (bofeng-song) [reply native/cocos/editor-support/spine-wasm/spine-wasm.cpp:98]: queryStoreMemory is responsible for allocation, while freeStoreMemory is responsible for deallocation.

Comment (minggo) [reply native/cocos/editor-support/spine-wasm/spine-wasm.cpp:98]: If so, why not rename to createStoreMemory?

Comment (bofeng-song) [reply native/cocos/editor-support/spine-wasm/spine-wasm.cpp:98]: Fixed

**Review by minggo — APPROVED** [review]

**Pull request merged**